src/usage-examples.test.ts

```
@@ -0,0 +1,553 @@
+        }
+
+        canHandle(value: any): boolean {
+          return value instanceof Date;
+        }
+      }
+
+      // Note: This would require RunCache.addSerializationAdapter method
+      // For now, we'll test the adapter directly
+      const adapter = new DateSerializationAdapter();
+      const testDate = new Date('2023-12-25T10:30:00Z');
+      
```

ESLint

[error] Trailing spaces not allowed.

(no-trailing-spaces)